Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.cpp

```diff
@@ -45,17 +45,34 @@ bool RemoteInspectionTarget::remoteControlAllowed() const
 
 bool RemoteInspectionTarget::inspectable() const
 {
-#if PLATFORM(COCOA)
-    static bool allowInternalSecurityPolicies = os_variant_allows_internal_security_policies("com.apple.WebInspector");
-    if (allowInternalSecurityPolicies && !RemoteInspector::singleton().isSimulatingCustomerInstall())
+    switch (m_inspectable) {
+    case Inspectable::Yes:
         return true;
+    case Inspectable::No:
+#if PLATFORM(COCOA)
+        static bool allowInternalSecurityPolicies = os_variant_allows_internal_security_policies("com.apple.WebInspector");
```

Comment:
Should the code in this switch case be done inside of setInspectable? That way, maybe we don't need this new enum with three values, and can keep using a bool?

Reply:
Thank you for the feedback! Halfway through responding to this suggestion I realized we have a bug that will likely require us to do this work in the getter.

There is actually a bug here currently that occurs when internal policies are being applied: the APIs for WKWebView and JSContext will return `true` even for contexts that have disabled inspection at the API level, meaning the setting of `inspectable` on those objects won't appear to do anything on system following an internal policy, which could lead to confusing or undesirable behavior in applications. The internal policies are meant to be invisible to clients of the libraries, but currently we are technically leaking this implementation detail.

Really what we should have is an `inspectable` getter like we have today that ignores internal policies, and an extra `allowsInspection` that actually takes internal policy into account, and is used by RemoteInspector* to determine the actual inspectability of the `RemoteInspectionTarget`.

For what its worth, the existing `bool` couldn't have been packed with any other bits anyway in this class, so the size of `RemoteInspectionTarget` should remain the same moving from `bool` to a `uint8_t` enum.